lib/src/main/java/org/openapitools/openapistylevalidator/UserDefinedRuleExecutor.java

```diff
@@ -0,0 +1,35 @@
+package org.openapitools.openapistylevalidator;
+
+import org.eclipse.microprofile.openapi.models.OpenAPI;
+import org.openapitools.openapistylevalidator.custom.OpenAPIRule;
+import org.reflections.Reflections;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Set;
+
+import static org.reflections.scanners.Scanners.SubTypes;
+
+public class UserDefinedRuleExecutor {
+
+    private ErrorAggregator errorAggregator;
+    private OpenAPI openAPI;
+
+    public UserDefinedRuleExecutor(ErrorAggregator errorAggregator, OpenAPI openAPI) {
+        this.errorAggregator = errorAggregator;
+        this.openAPI = openAPI;
+    }
+
+    public void loadAndExecuteRules() throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException {
```

Comment:
This is interesting but it would make everyone using the library to use the "custom" rules that you created. I think maybe each rules should implement a "uniqueName". Then, in the CLI or maven/gradle, people could pass an array of the rules they want to check. In this step, we would check if the rules is in the list and only execute the one in the list.

In the future, if that work great, we could create a concept of "playlist" or something like that where people don't have to submit a big list of rules but just the name of a playlist containing a list of rules. Or continue to use the option files too will work :)

Reply:
Yes.. that's a good idea. We will assign a unique name to each rule in this version and as next step we can look for playlist concept as you suggested 👍